src/v/redpanda/application.cc

```diff
@@ -104,7 +104,11 @@ static void set_local_kafka_client_config(
   const config::node_config& config) {
     client_config.emplace();
     const auto& kafka_api = config.kafka_api.value();
-    vassert(!kafka_api.empty(), "There are no kafka_api listeners");
+    if (kafka_api.empty()) {
+        // No Kafka listeners configured, cannot configure
+        // a client.
+        return;
```

Comment:
this would preclude configurations where pandaproxy/schema-registry were running but the node wasn't operating as a kafka broker--we'd have to come up with some other means to seed the broker list. is that intentional?

Reply:
The intended use is to run a whole cluster without kafka listeners early during setup, rather than to selectively have some nodes acting as SR/PP only.

I guess there is a downside that we're allowing a node_config that is accepted (no kafka listeners but sr/pp listeners) but that doesn't actually work for user traffic that hits SR/PP.

We could wait for controller replay before constructing PP/SR config, but that would only help in the case that there were some other nodes listening on Kafka ports (and not in the intended case of a primordial cluster that isn't doing any kafka yet).  That could be an interesting follow-on: to reinstatea the pseudo-standalone PP/SR mode as a mode where a node has no kafka listeners (we'd have to make it also never try to be leader of anything) but participates in the controller group.

Reply:
>I guess there is a downside that we're allowing a node_config that is accepted (no kafka listeners but sr/pp listeners) but that doesn't actually work for user traffic that hits SR/PP.

It seems ok. If user traffic hits SR/PP a reasonable error should be returned indicating a lack of connectivity, although i don't know exactly which error that would be.